### DISCUSSION TITLE
Remove "Click to disable this" from unique names '%'

### DISCUSSION THREAD

**Discussion by nubels** [godotengine/godot-proposals#8678] [Editor]
I use the unique names feature a lot and I think it really awesome but I often revoke the unique name by accident because I missclick on the '%' sign. Can anyone relate?
[edited]
I think this behaviour can be a annoying, especially because revoking unique names can break the functionality of scripts.

That's why I think "Click to disable this" should be removed from unique names '%', or there should be an option to disable this behaviour.

![Godot_v4 2 1-stable_win64_6kGj3ak2qM](https://github.com/godotengine/godot-proposals/assets/81755541/e31b0dca-f693-4984-9210-d31caa6fbba4)

**Comment by AThousandShips** [comment]
You're missing several other icons that *do* change behaviour, like the visibility icon, and the lock (which turns of locking when clicked), so it's far from no other cases like this

Buttons that directly deactivate the thing they represent:
* Unique name
* Lock
* Group
* Pin (for animation)

That's in fact just short of *half* of all the icons (actually half if you include visibility)

**Reply by nubels** [reply]
Ah okay, I see. My bad for missing that. But I think the use case for unique names is different from what the buttons you mentioned do. You can write scripts that depend on unique names and it's very easy to break them by revoking the unique name by accident. 
If I'm not mistaken, removing locks, groups, etc does not result in breaking changes. That's why I still think revoking unique names should not be a one click workflow.

**Reply by AThousandShips** [reply]
But removing unique name doesn't cause any problems as long as you apply it again, you don't lose any information, it's annoying to click wrong sure, but it doesn't break anything unless you forget to add it back

**Reply by nubels** [reply]
So you do agree with my point that it's annoying. That's what I wanted to convey in the first place. I'm sorry for making this more complicated than it needs to be.

**Reply by AThousandShips** [reply]
But you give it more weight than it needs... And I didn't say I'm annoyed by this, because if doesn't happen to me, I said I understand that accidentally clicking this is annoying

However having to open a menu to disable it is *also* annoying and is a greater reduction of usability IMO than having to fix it if you make a mistake

**Reply by nubels** [reply]
Ah okay. I suppose we would both be satisfied if there was an option to disable this behaviour that was turned off by default. Let's see what the community thinks.

**Comment by Mickeon** [comment]
I think the overall presentation of unique nodes in the Scene Dock is lackluster and contributes to icon clutter as, it currently is. The fact that you can "disable" it on click is only part of the issues.

- https://github.com/godotengine/godot-proposals/issues/4494

**Comment by Kakiroi** [comment]
This is absolute nightmare for me. Because it doesn't even tell you that unique name is turned off(and not obvious), I might accidentally turn it off and script breaks, have to go for bug hunting. For people who have smaller screen, this will be even worse, because how small scene dock would be, even trying to just click node will accidentally trigger that button. 
Turning unique name on is very specific operation(right click, click on operation). Why turning it off is so easy? I would rather have it other way around. At least you won't make critical mistake that way.
